backend/src/controllers/user.js

```diff
@@ -394,4 +395,25 @@ export default {
       next(err);
     }
   },
+
+  /**
+   * Express.js controller for POST /users/choose-fav-deck/:id
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   * @param {import('express').NextFunction} next
+   * @returns {Promise<void>}
+   */
+  chooseFavDeck: async (req, res, next) => {
+    try {
+      const deck = await deckService.findById(req.params.id);
+      if (!deck) { return res.sendStatus(404); }
+      if (deck.userId !== req.user.id) { return res.sendStatus(403); }
```

Comment:
```suggestion
      if (!deck) throw new Error('Deck not found', { cause: 'Not Found' });
      if (req.user.id !== deck.userId) throw new Error('You don't own this deck', { cause: 'Unauthorized' });
```

https://github.com/ESGI-69/challenge-node-vue/blob/51392b282c6e93d101e2305b15aa72ec5e1b5ea7/backend/src/errorHandler.js#L56-L59